clients/drcachesim/tools/reuse_distance.h

```diff
@@ -101,10 +104,22 @@ class reuse_distance_t : public analysis_tool_t {
                      unsigned int distance_limit, bool verify);
         std::unordered_map<addr_t, line_ref_t *> cache_map;
         std::unordered_set<addr_t> pruned_addresses;
-        // This is our reuse distance histogram.
-        std::unordered_map<int_least64_t, int_least64_t> dist_map;
+        // These are our reuse distance histograms: one for all accesses and one
+        // only for data references.  An instruction histogram can be computed by
+        // subtracting data references from the full histogram.  The final histogram
+        // statistics need a full histogram to sort, and in most traces the majority of
+        // accesss are instruction references, so the histograms are split this way to
+        // provide the full histogram we need with the smallest secondary histogram.
+        // Furthermore, during analysis each reference is added to only one histogram
+        // to minimize the performance impact of dual histogram collection, with the data
+        // references added into the primary histogram during final result aggregation.
+        // This means dist_map is effectively instruction-only until aggregation.
```

Comment:
Hmm, this has potential to cause confusion: I guess we don't want to pay the space for two separate histograms, dist_map_instr and dist_map_combined?

Reply:
Yes, I debated this, and agree it's a bit confusing.  I started going down the path of having instruction-only and data-only histograms.  But during the final summary and aggregation stages there are multiple uses of the unified histogram, so it either needed to be created and kept around as a third histogram, or else the instruction-only histogram needed to be repurposed.  I guess my solution is the latter, except I named the dual-use histogram by its final form, rather then the as-collected form.  I'll keep it this way unless you feel strongly.